addons/account/models/account_reconcile_model.py

```diff
@@ -831,6 +831,10 @@ def get_order_by_clause(alias=None):
                     'allow_auto_reconcile': True,
                     'amls': self.env['account.move.line'].browse(candidate_ids),
                 }
+            elif self.match_text_location_label or self.match_text_location_note or self.match_text_location_reference:
+                # In the case any of the Label, Note or Reference matching rule has been toggled, and the query didn't return
+                # any candidates, the model should not try to mount another aml instead.
```

Comment:
Just discussed it with CHKL, which agrees with what I have now : if nothing is toggled but "Partner is Set", we should not try to match with anything, and mount any correct line from that partner. If one or more are toggled, we should mount *only* a matching line, or no line at all if none is found.
(I responded to be sure not to forget, but we'll wait for your return in case you want to challenge that point of view)

Reply:
@proose remove the fallback below instead